core/src/main/java/org/opensearch/sql/storage/StorageEngine.java

```diff
@@ -18,14 +19,14 @@ public interface StorageEngine {
   /**
    * Get {@link Table} from storage engine.
    */
-  Table getTable(String name);
+  Table getTable(CatalogSchemaName catalogSchemaName, String tableName);
 
   /**
    * Get list of catalog related functions.
    *
    * @return FunctionResolvers of catalog functions.
    */
-  default Collection<FunctionResolver> getFunctions() {
+  default Collection<FunctionResolver> getFunctions(CatalogSchemaName catalogSchemaName) {
```

Comment:
1. Functions belong to Catalog not schema.
2. Catalog create StorgeEngine. `catalog.storageEngine` which means StorageEgnine already have info of catalog. Then we do need treat catalogSchema as params.